web_src/js/utils/dom.js

```diff
@@ -59,6 +59,15 @@ export function onDomReady(cb) {
   }
 }
 
+export function isDocumentFragmentOrElementNode(el) {
+  try {
+    return el.nodeType === Node.ELEMENT_NODE || el.nodeType === Node.DOCUMENT_FRAGMENT_NODE;
+  } catch {
+    // in case the el is not in the same origin, then the access to nodeType would fail
+    return false;
+  }
+}
+
```

Comment:
Could we detect iframed document via `el.ownerDocument === document` and ignore those? I don't think we want to run the mutation callbacks on any iframed content (mermaid and user-configured renderers are the only cases I know of).

Reply:
done in 6299b57fd608bdd18781e5457115079ff568463c , and added more comments